windows-containers/cluster-logging-windows-worker-nodes.md

```diff
@@ -0,0 +1,174 @@
+---
+title: OpenShift Windows Worker Node and Container Logging
+authors:
+  - "@ravisantoshgudimetla"
+  - "@aravindhp"
+reviewers:
+  - "@crawford"
+  - "@sdodson"
+  - "@jcantrill"
+  - "@richm" 
+approvers:
+  - "@crawford"
+  - "@sdodson"
+  - "@jcantrill"
+  - "@richm"  
+creation-date: 2019-09-10
+last-updated: 2019-09-17
+status: implementable
+---
+
+# logging-for-windows-worker-nodes
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+The intent of this enhancement is to enable logging for Windows worker nodes 
+and pods running in an OpenShift cluster
+
+## Motivation
+
+Logging is a critical component for identifying issues with nodes, containers
+running on the nodes. The main motivation behind this enhancement is to bring 
+Windows node logging infrastructure on par with linux nodes
+
+### Goals
+
+As part of this enhancement, we plan to do the following:
+* Deploy log collection infrastructure onto Windows nodes
+* Collect node and pod logs
+* Upgrade log collection infrastructure
+* Forward logs and events from nodes and pods to log store
+
+### Non-Goals
+
+As part of this enhancement, we do not plan to support:
+* Logging for Windows containers
+  * Microsoft will provide us with the logging framework that enables Windows 
+    containers to redirect logs to standard out from various sources like 
+    ETW, Perf Counters etc
+* Customization of log collection infrastructure
+  * We plan to support fluentd only
+  * Microsoft will provide us with a Windows fluentd binary or gem file
+
+
+## Proposal
+
+The main idea here is to deploy fluentd as a log collection agent onto Windows 
+nodes after cluster logging operator is deployed onto the cluster. Fluentd has
+support for Windows logging. We plan to leverage it and run fluentd as a 
+Windows service.
+
+
+### Implementation Details
+
+We plan to create a Windows Node logging Ansible Playbook that can
+
+* Download fluentd binary for Windows from a known location and transfers it to
+  the Windows node
+* Configure the container runtime to use fluentd as the logging driver
+* Configure fluentd to communicate with elasticsearch data store in the same 
```

Comment:
For example, here is a playbook we used to grab the fluentd secret from an existing cluster in 3.11: https://github.com/openshift/origin-aggregated-logging/blob/release-3.11/hack/testing/rsyslog/playbook.yaml - see especially https://github.com/openshift/origin-aggregated-logging/blob/release-3.11/hack/testing/rsyslog/playbook.yaml#L45-L58 - note that in 4.x, the secret is `secret/fluentd` - no `logging-` prefix on the name